From 470d9385f7c772c4cb88e0fdfa3484baba9d15c9 Mon Sep 17 00:00:00 2001
From: Thomas Glanzmann <thomas@glanzmann.de>
Date: Mon, 21 Feb 2022 23:01:36 +0100
Subject: [PATCH 012/171] arm64: dts: apple: Add backlight node to j293/j313

It can be turned off with:

echo 1 > /sys/class/backlight/backlight/bl_power

It can be turned on with:

echo 0 > /sys/class/backlight/backlight/bl_power

Needs CONFIG_BACKLIGHT_GPIO=m.

Signed-off-by: Thomas Glanzmann <thomas@glanzmann.de>
---
 arch/arm64/boot/dts/apple/t8103-j293.dts | 12 ++++++++++++
 arch/arm64/boot/dts/apple/t8103-j313.dts | 12 ++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/apple/t8103-j293.dts b/arch/arm64/boot/dts/apple/t8103-j293.dts
index 4f5f7d38e799..6f1050185a80 100644
--- a/arch/arm64/boot/dts/apple/t8103-j293.dts
+++ b/arch/arm64/boot/dts/apple/t8103-j293.dts
@@ -75,3 +75,15 @@ &i2c2 {
 &i2c4 {
 	status = "okay";
 };
+
+/ {
+	backlight: gpio-bl {
+		compatible = "gpio-backlight";
+		gpios = <&smc_gpio 18 GPIO_ACTIVE_HIGH>;
+		default-on;
+	};
+};
+
+&framebuffer0 {
+	backlight = <&backlight>;
+};
diff --git a/arch/arm64/boot/dts/apple/t8103-j313.dts b/arch/arm64/boot/dts/apple/t8103-j313.dts
index d6722d8809d1..3e4045c05905 100644
--- a/arch/arm64/boot/dts/apple/t8103-j313.dts
+++ b/arch/arm64/boot/dts/apple/t8103-j313.dts
@@ -67,3 +67,15 @@ &pcie0_dart_2 {
 
 /delete-node/ &port01;
 /delete-node/ &port02;
+
+/ {
+	backlight: gpio-bl {
+		compatible = "gpio-backlight";
+		gpios = <&smc_gpio 18 GPIO_ACTIVE_HIGH>;
+		default-on;
+	};
+};
+
+&framebuffer0 {
+	backlight = <&backlight>;
+};
-- 
2.34.1

